File: wayland/src/main/java/org/westmalle/wayland/nativ/libdrm/DrmEventContext.java
package org.westmalle.wayland.nativ.libdrm;

import org.freedesktop.jaccall.CType;
import org.freedesktop.jaccall.Field;
import org.freedesktop.jaccall.Struct;

@Struct({
                @Field(name = "version",
                       type = CType.INT),
                @Field(name = "vblank_handler",
                       type = CType.POINTER,
                       dataType = vblank_handler.class),
                @Field(name = "page_flip_handler",
                       type = CType.POINTER,
                       dataType = page_flip_handler.class)
        })
public final class DrmEventContext extends DrmEventContext_Jaccall_StructType {}

File: compositor/src/main/kotlin/org/westford/nativ/libdrm/DrmEventContext.kt
/*
 * Westford Wayland Compositor.
 * Copyright (C) 2016  Erik De Rijcke
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.westford.nativ.libdrm

import org.freedesktop.jaccall.CType
import org.freedesktop.jaccall.Field
import org.freedesktop.jaccall.Struct

@Struct(Field(name = "version",
              type = CType.INT),
        Field(name = "vblank_handler",
              type = CType.POINTER,
              dataType = vblank_handler::class),
        Field(name = "page_flip_handler",
              type = CType.POINTER,
              dataType = page_flip_handler::class)) class DrmEventContext : Struct_DrmEventContext()
